src/pixelator/graph/backends/implementations.py

```diff
@@ -6,12 +6,15 @@
 from __future__ import annotations
 
 import logging
-from typing import Dict, Iterable, List, Optional, Tuple
+import warnings
+from typing import Dict, Iterable, List, Optional, Tuple, Union
 
 import igraph
 import networkx as nx
 import numpy as np
 import pandas as pd
+import polars as pl
+from networkx.algorithms import bipartite
```

Comment:
I would like to make more explicit that `bipartite` is  coming from `networkx` so it is easier to separate igraph from networkx